packages/desktop-client/src/components/reports/spreadsheets/filterHiddenItems.ts

```diff
@@ -0,0 +1,23 @@
+import {
+  type QueryDataEntity,
+  type UncategorizedEntity,
+} from '../ReportOptions';
+
+type filterHiddenItemsProps = {
+  item: UncategorizedEntity;
+  data: QueryDataEntity[];
+};
+
+function filterHiddenItems({ item, data }: filterHiddenItemsProps) {
+  let filtered = data.filter(asset =>
+    item.uncat_id
+      ? (item.transfer ? asset.transferAccount : !asset.transferAccount) &&
+        (item.category ? true : !asset.category) &&
+        (item.offBudget ? asset.accountOffBudget : !asset.accountOffBudget)
+      : true,
+  );
+
+  return filtered;
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏
```suggestion
  return data.filter(
      asset => {
          if (!item.uncat_id) {
              return true;
          }
          
          // TODO: variable naming could definitely be improved; I just came up with some example vars
          const isTransferableAccount = (item.transfer ? asset.transferAccount : !asset.transferAccount);
          const shouldItemBeHidden = (item.category ? true : !asset.category);
          const isOffBudget = (item.offBudget ? asset.accountOffBudget : !asset.accountOffBudget);
          
          return isTransferableAccount && shouldItemBeHidden && isOffBudget;
      }
  );
```